python/lsst/sims/catUtils/mixins/VariabilityMixin.py

```diff
@@ -386,56 +488,298 @@ def applyBHMicrolens(self, params, expmjd_in):
+                                   'lsstBandpassDict being defined.')
+
+            ebv_grid = numpy.arange(0.0, 7.01, 0.01)
+            bb_wavelen = numpy.arange(200.0, 1500.0, 0.1)
+            hc_over_k = 1.4387e7  # nm*K
+            temp = 9000.0  # black body temperature in Kelvin
+            exp_arg = hc_over_k/(temp*bb_wavelen)
+            exp_term = 1.0/(numpy.exp(exp_arg) - 1.0)
+            ln_exp_term = numpy.log(exp_term)
+
+            log_bb_flambda = -5.0*numpy.log(bb_wavelen) + ln_exp_term
+            bb_flambda = numpy.exp(log_bb_flambda)
+            bb_sed = Sed(wavelen=bb_wavelen, flambda=bb_flambda)
+
+            base_fluxes = self.lsstBandpassDict.fluxListForSed(bb_sed)
+
+            a_x, b_x = bb_sed.setupCCMab()
+            self._mlt_dust_lookup = {}
+            self._mlt_dust_lookup['ebv'] = ebv_grid
+            list_of_bp = self.lsstBandpassDict.keys()
+            for bp in list_of_bp:
+                self._mlt_dust_lookup[bp] = numpy.zeros(len(ebv_grid))
+            for iebv, ebv_val in enumerate(ebv_grid):
+                wv, fl = bb_sed.addCCMDust(a_x, b_x,
+                                           ebv=ebv_val,
+                                           wavelen=bb_wavelen,
+                                           flambda=bb_flambda)
+
+                dusty_bb = Sed(wavelen=wv, flambda=fl)
+                dusty_fluxes = self.lsstBandpassDict.fluxListForSed(dusty_bb)
+                for ibp, bp in enumerate(list_of_bp):
+                    self._mlt_dust_lookup[bp][iebv] = dusty_fluxes[ibp]/base_fluxes[ibp]
+
+        # get the distance to each star in parsecs
+        _au_to_parsec = 1.0/206265.0
+        dd = _au_to_parsec/parallax
+
+        # get the area of the sphere through which the star's energy
+        # is radiating to get to us (in cm^2)
+        _cm_per_parsec = 3.08576e16
+        sphere_area = 4.0*numpy.pi*numpy.power(dd*_cm_per_parsec, 2)
+
+        flux_factor = self.photParams.effarea/sphere_area
+
+        dMags = numpy.zeros((6, self.num_variable_obj(params)))
+        mag_name_tuple = ('u', 'g', 'r', 'i', 'z', 'y')
+        base_fluxes = {}
+        base_mags = {}
+        ss = Sed()
+        for mag_name in mag_name_tuple:
+            if ('lsst_%s' % mag_name in self._actually_calculated_columns or
+                'delta_lsst_%s' % mag_name in self._actually_calculated_columns):
+
+                mm = self.column_by_name('quiescent_lsst_%s' % mag_name)
+                base_mags[mag_name] = mm
+                base_fluxes[mag_name] = ss.fluxFromMag(mm)
+
+        lc_name_arr = params['lc'].astype(str)
+        lc_names_unique = numpy.unique(lc_name_arr)
+        for lc_name in lc_names_unique:
+            if 'None' in lc_name:
+                continue
+
+            use_this_lc = numpy.where(numpy.char.find(lc_name_arr, lc_name)==0)
+
+            lc_name = lc_name.replace('.txt', '')
+            if 'late' in lc_name:
```

Comment:
There isn't supposed to be a 'late_inactive' class, but I was stupid an assigned some 'late_inactive' stars n fatboy.  Rather than fix the fatboy curves (which will take about a week of compute time), I figured I'd fix it here.